### PR TITLE
tgui port: proxy storage

### PR DESCRIPTION
**Pull request by ZeWaka**
## About the PR 

ports https://github.com/tgstation/tgstation/pull/53294

## Why's this needed? 

mmm tasty bugfixes yummy

### CODE REVIEW

File: tgui/packages/common/storage.js [hunk 1]
```diff
@@ -150,8 +150,47 @@ class IndexedDbBackend {
...
+class StorageProxy {
+  constructor() {
+    this.backendPromise = (async () => {
+      if (testIndexedDb()) {
+        try {
+          const backend = new IndexedDbBackend();
+          await backend.dbPromise;
+          return backend;
+        }
+        catch {}
```
Comment (mordent-goonstation) [inline tgui/packages/common/storage.js:166]: Silent catch bad.

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: tgui/packages/common/storage.js (hunk 1 above)
Comment (ZeWaka) [reply tgui/packages/common/storage.js:166]: upstream issue?

Comment (mordent-goonstation) [reply tgui/packages/common/storage.js:166]: Seems so.

Comment (mordent-goonstation) [reply tgui/packages/common/storage.js:166]: @stylemistake Is this something you'd want us to fix upstream?

**Review by mordent-goonstation — APPROVED** [review]

File: tgui/packages/common/storage.js (hunk 1 above)
Comment (ZeWaka) [reply tgui/packages/common/storage.js:166]: upstream reasoning: below lines and `It falls back gracefully to LocalStorageBackend, and MemoryBackend if browser throws an error in the process.`

**Pull request merged**